### PR TITLE
add linter for script and mocha test file

### PR DESCRIPTION
**Pull request by hueitan**
use [eslint-config-wikimedia](https://github.com/wikimedia/eslint-config-wikimedia) as the linter

### CODE REVIEW

**Comment by stephanebisson** [comment]
I would like to propose that we disable the rule that forces semi-colons. Unlike MediaWiki, our code is transpiled and semi-colons are added automatically.

**Comment by hueitan** [comment]
> I would like to propose that we disable the rule that forces semi-colons. Unlike MediaWiki, our code is transpiled and semi-colons are added automatically.

True, we don't need `;` in our code. rule added.

Regards the mocha test rule, I don't modify it as we follow the config from wikimedia, what do you say?

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "test": "mocha --require ignore-styles,@babel/register",
     "test:watch": "mocha --require ignore-styles,@babel/register -w",
+    "test:fix-lint": "eslint test --fix",
```
Comment (stephanebisson) [inline package.json:9]: I don't think this task should be called "test"... maybe just "lint" or "eslint"

Comment (hueitan) [reply package.json:9]: This is the linter for the folder test, and it has it’s owned eslint config, do you think we also need to apply no semi in the test spec?

Note that client code and test code use different eslint config

Comment (stephanebisson) [reply package.json:9]: Oh, I didn't notice that. Why do they have different configs?

Comment (hueitan) [reply package.json:9]: There is config for client code, 

and one for mocha, i think that’s the reason mocha has the variable describe and it, and runs with node script

https://github.com/wikimedia/eslint-config-wikimedia

**Comment by stephanebisson** [comment]
> Regards the mocha test rule, I don't modify it as we follow the config from wikimedia, what do you say?

I don't like having semis in one part of the code and not in the other. I would prefer to omit them everywhere.

**Comment by hueitan** [comment]
> > Regards the mocha test rule, I don't modify it as we follow the config from wikimedia, what do you say?
> 
> I don't like having semis in one part of the code and not in the other. I would prefer to omit them everywhere.

Done 💪

**Pull request merged**